engine/crash/src/backtrace_libunwind.cpp

```diff
@@ -197,13 +198,31 @@ namespace dmCrash
         OnCrash(0xDEAD);
     }
 
+    static void ResetToDefaultHandler(const int signum)
+    {
+        struct sigaction sa;
+        memset(&sa, 0, sizeof(sa));
+        sigemptyset(&sa.sa_mask);
+        sa.sa_handler = SIG_DFL;
+        sa.sa_flags = 0;
+        sigaction(signum, &sa, NULL);
+    }
+
     static void Handler(const int signum, siginfo_t *const si, void *const sc)
     {
+        // If after set of a handler from the previous (default)
+        // behavior from sigdfl we still enter this function
+        // we reset handler to default to prevent infinite loop
+        if (g_AlreadyHandled[signum]) {
+            ResetToDefaultHandler(signum);
```

Comment:
How does this differ from the code on line 223 ?
And, why not change the code there?

Reply:
as I understand we try to restore prev. behaviour, previous doesn't necessary means default if some extension use it's own `sigaction` for example
so, I decided to give a chance to restore previous behaviour, if it fails, then I reset to default

Reply:
It says "default" in the comment?
And, the comment also refers to the exact looping issue you're trying to solve?
So, why solve it in two places?

Also note that since you introduced the bug with uninitialized memory, it's entirely possible you hit this it-statement without ever calling OnCrash() below.

Reply:
it says `// The previous (default)`, but ok, make sense, I'll just reset it to default then

Reply:
Now I reset to default behaviour without trying to restore previous behaviour.